import/index.html

```diff
@@ -0,0 +1,298 @@
+        return [...buffer]
+          .map(x => x.toString(16).padStart(2, '0'))
+          .join('');
+      }
+
+      /**
+       * Additional Associated Data (AAD) in the format dictated by the enclave_encrypt crate.
+       */
+      const additionalAssociatedData = (senderPubBuf, receiverPubBuf) => {
+        const s = Array.from(new Uint8Array(senderPubBuf));
+        const r = Array.from(new Uint8Array(receiverPubBuf));
+        return new Uint8Array([...s, ...r]);
+      }
+
+      /**
+       * Function to send a message. If this page is embedded as an iframe we'll use window.top.postMessage. Otherwise we'll display it in the DOM.
+       * @param type message type. Can be "PUBLIC_KEY_CREATED" or "BUNDLE_INJECTED"
+       * @param value message value
+       */
+      const sendMessageUp = (type, value) => {
+        if (window.top !== null) {
+          window.top.postMessage({
+              "type": type,
+              "value": value,
+          }, '*')
+        }
+      }
+
+
+      return {
+        importTargetKey,
+        setItemWithExpiry,
+        getItemWithExpiry,
+        getTargetEmbeddedKey,
+        setTargetEmbeddedKey,
+        sendMessageUp,
+        uint8arrayFromHexString,
+        uint8arrayToHexString,
+        additionalAssociatedData
+      }
+    }();
+  </script>
+
+  <!--
+    Script importing HPKE lib until we can replace it
+    Because this is loaded as a module JSDOM can't load it properly
+    Code in here isn't tested, so let's keep this to a minimum!
+  -->
+  <script type="module">
+    // TODO: this should be bundled at build time or replaced with code written by Turnkey entirely.
+    import * as hpke from "https://esm.sh/@hpke/core";
+
+    document.addEventListener("DOMContentLoaded", async () => {
+      // This is a workaround for how the SDK's iframe stamper is initialized. Currently,
+      // init() waits for a public key to be initialized that can be used to send to the server
+      // which will encrypt messages to this public key.
+      // In the case of import, this public key is not used because the client encrypts messages
+      // to the server's public key.
+      TKHQ.sendMessageUp("PUBLIC_KEY_READY", "")
+      // TODO: find a way to filter messages and ensure they're coming from the parent window?
+      // We do not want to arbitrarily receive messages from all origins.
+      window.addEventListener("message", async function(event) {
+        if (event.data && event.data["type"] == "INJECT_IMPORT_BUNDLE") {
+          try {
+            await onInjectImportBundle(event.data["value"])
+          } catch (e) {
+            TKHQ.sendMessageUp("ERROR", e.toString());
+          }
+        }
+        if (event.data && event.data["type"] == "EXTRACT_WALLET_ENCRYPTED_BUNDLE") {
+          try {
+            await onExtractWalletEncryptedBundle()
+          } catch (e) {
+            TKHQ.sendMessageUp("ERROR", e.toString());
+          }
```

Comment:
I wonder if we should delete this interface and have the iframe automatically send an event "WALLET_ENCRYPTED" with the encrypted content whenever encryption succeeds? That way the parent page injects the import bundle, and then waits for this WALLET_ENCRYPTED event.

Basically encryption should happen when the user presses enter or submits the form that's within the iframe, and the parent page doesn't have to know when this happens. But I don't know if that's a reasonable thing: the parent page may want to have its own "submit" button 🤷

Reply:
oh i like this idea! this flow with a button inside the iframe potentially makes it 2 clicks for the user (click button in iframe then passkey tap for the api call made outside of the iframe)... so wdyt about supporting this as a "styling/settings" option in my follow up or for settings (basically an event that this page listens to for setting which content shows up and what styling to apply to it via a json)